source/class/qx/ui/treevirtual/SimpleTreeDataModel.js

```diff
@@ -1099,6 +1104,76 @@ qx.Class.define("qx.ui.treevirtual.SimpleTreeDataModel", {
       return (
         this.__tree.getAllowNodeEdit() && this.getNodeFromRow(rowIndex).bCanEdit
       );
+    },
+
+    /**
+     * Expands all nodes in the tree with minimal redraw
+     */
+    expandAll() {
```

Comment:
What do you think of moving the changes from here onwards into MNode.js, which already has similar utility methods such as `nodeToggleOpened`? If doing so, `collapseBeneath` could be renamed to match the naming in MNode, to `nodeCollapseBeneath`. Similarly, `expandBeneath` -> `nodeExpandBeneath`.

Reply:
In fact, there's already a `nodeSetOpened`. Maybe add a `recursive` parameter to it, in lieu of `nodeCollapseBeneath` and `nodeExpandBeneath`? The only question, both in this case and in your original, is whether the specified node itself should be collapsed or expanded. Just need to make sure it's documented what it does.